src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx

```diff
@@ -141,38 +184,143 @@ describe("CreateWallet", () => {
+			await waitFor(() => expect(rendered.getByTestId(`CreateWallet__first-step`)).toBeTruthy());
 
 			fireEvent.click(continueButton);
-			expect(getByTestId(`CreateWallet__third-step`)).toBeTruthy();
+			await waitFor(() => expect(getByTestId(`CreateWallet__second-step`)).toBeTruthy());
 
 			fireEvent.click(continueButton);
+			await waitFor(() => expect(getByTestId(`CreateWallet__third-step`)).toBeTruthy());
+
+			const walletMnemonic = passphrase.split(" ");
+			for (let i = 0; i < 3; i++) {
+				const wordNumber = getByText(/Select word #/).innerHTML.replace(/Select word #/, "");
 
-			// Submit
+				fireEvent.click(getByText(walletMnemonic[wordNumber - 1]));
 
-			fireEvent.click(getByTestId(`CreateWallet__save-button`));
+				if (i < 2) {
+					await waitFor(async () => (await findAllByText(/The #([0-9]+) word/)).length === 2 - i);
+				}
+			}
+
+			await waitFor(() => expect(continueButton).not.toHaveAttribute("disabled"));
+
+			fireEvent.click(continueButton);
+			await waitFor(() => expect(getByTestId(`CreateWallet__fourth-step`)).toBeTruthy());
+
+			await fireEvent.change(getByTestId("CreateWallet__wallet-name"), { target: { value: "Test Wallet" } });
+			await fireEvent.click(getByTestId(`CreateWallet__save-button`));
+
+			await waitFor(() =>
+				expect(profile.wallets().values()[0].settings().get(WalletSetting.Alias)).toEqual("Test Wallet"),
+			);
+		});
+
+		expect(asFragment()).toMatchSnapshot();
+	});
+
+	it("should not have a pending wallet if leaving on step 1", async () => {
+		let rendered: RenderResult;
+		const history = createMemoryHistory();
+		const createURL = "/profiles/bob/wallets/create";
+		history.push(createURL);
+
+		await act(async () => {
+			rendered = renderWithRouter(
+				<EnvironmentProvider env={env}>
+					<Route path="/profiles/:profileId/wallets/create">
+						<CreateWallet />
+					</Route>
+				</EnvironmentProvider>,
+				{
+					routes: [createURL, "/"],
+					history,
+				},
+			);
+
+			await waitFor(() => expect(rendered.getByTestId(`CreateWallet__first-step`)).toBeTruthy());
+		});
+
+		const { getByTestId, asFragment } = rendered!;
+
+		history.push("/");
+		await waitFor(() => expect(profile.wallets().values().length).toBe(0));
+
+		expect(asFragment()).toMatchSnapshot();
+	});
+
+	it("should remove pending wallet if not submitted", async () => {
+		let rendered: RenderResult;
+		const history = createMemoryHistory();
+		const createURL = "/profiles/bob/wallets/create";
+		history.push(createURL);
+
+		await act(async () => {
+			rendered = renderWithRouter(
+				<EnvironmentProvider env={env}>
+					<Route path="/profiles/:profileId/wallets/create">
+						<CreateWallet />
+					</Route>
+				</EnvironmentProvider>,
+				{
+					routes: [createURL, "/"],
+					history,
+				},
+			);
+
+			await waitFor(() => expect(rendered.getByTestId(`CreateWallet__first-step`)).toBeTruthy());
+		});
+
+		const { getByTestId, getByText, asFragment } = rendered!;
```

Comment:
`getByText` is declared but its value is never read.

Reply:
Updated, thanks!